contracts/GnosisSafe.sol

```diff
@@ -140,12 +140,15 @@ contract GnosisSafe
             txHash = keccak256(txHashData);
             checkSignatures(txHash, txHashData, signatures, true);
         }
-        require(gasleft() >= safeTxGas, "Not enough gas to execute safe transaction");
+        // We require some gas to emit the events (2500) after the execution and some to perform code until the execution (500)
+        // We also increase the amount required by 1/64 to counteract potential shortings because of EIP-150
+        require(gasleft() - 3000 >= safeTxGas * 65 / 64, "Not enough gas to execute safe transaction");
```

Comment:
Shouldn't the gas for events be part of `baseGas`?

Reply:
BaseGas is a payment information, this check is to ensure that there is enough gas (independent of the payment)

Reply:
Ok then